crates/wasmtime/src/runtime.rs

```diff
@@ -775,12 +800,35 @@ impl Store {
         self.engine().config.memory_creator.as_ref().map(|x| x as _)
     }
 
-    pub(crate) fn compiler(&self) -> std::cell::Ref<'_, Compiler> {
-        self.inner.compiler.borrow()
+    pub(crate) fn signatures(&self) -> std::cell::Ref<'_, SignatureRegistry> {
+        self.inner.signatures.borrow()
+    }
+
+    pub(crate) fn signatures_mut(&self) -> std::cell::RefMut<'_, SignatureRegistry> {
+        self.inner.signatures.borrow_mut()
     }
 
-    pub(crate) fn compiler_mut(&self) -> std::cell::RefMut<'_, Compiler> {
-        self.inner.compiler.borrow_mut()
+    /// Returns whether or not the given address falls within the JIT code
+    /// managed by the compiler
+    pub(crate) fn is_in_jit_code(&self, addr: usize) -> bool {
+        self.inner
+            .jit_code_ranges
+            .borrow()
+            .iter()
+            .any(|(start, end)| *start <= addr && addr < *end)
+    }
+
+    pub(crate) fn register_jit_code(&self, mut ranges: impl Iterator<Item = (usize, usize)>) {
+        match ranges.next() {
+            None => (),
+            Some(first) => {
+                if !self.is_in_jit_code(first.0) {
```

Comment:
Could this have a comment for what's going on here? Without that it's a bit cryptic what this check is doing